tests/modeltests/tst_disassemblyoutput.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+  tst_disassemblyoutput.cpp
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2020 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Darya Knysh <d.knysh@nips.ru>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <QDebug>
+#include <QObject>
+#include <QTest>
+#include <QStandardPaths>
+#include <QProcess>
+#include <QString>
+#include <QStringList>
+#include <QVector>
+
+#include <models/disassemblyoutput.h>
+#include "data.h"
+
+class TestDisassemblyOutput : public QObject
+{
+    Q_OBJECT
+private slots:
+    void testFiles_data()
+    {
+        QTest::addColumn<QString>("binaryFile");
+
+        const auto files = {
+            "vector_static_gcc/vector_static_gcc_v9.1.0"
+        };
+        for (auto file : files)
+            QTest::addRow("%s", file) << file;
+    }
+
+    void testFiles()
+    {
+        QFETCH(QString, binaryFile);
+
+        const auto actualBinaryFile = QFINDTESTDATA(binaryFile);
+        QVERIFY(!actualBinaryFile.isEmpty() && QFile::exists(actualBinaryFile));
+        const auto expectedOutputFile = actualBinaryFile + ".expected.txt";
+        const auto actualOutputFile = actualBinaryFile + ".actual.txt";
+
+        Data::Symbol curSymbol = {QLatin1String("__cos_fma"),
```

Comment:
pass the symbol in as another QTest column, then only patch/overwrite the actualBinaryFile here

Reply:
also: here and below these should be QStringLiteral instead of QLatin1String, or remove it altogether - in a test that's OK imo